fastlane/lib/fastlane/actions/setup_circle_ci.rb

```diff
@@ -24,15 +24,20 @@ def self.setup_output_paths(params)
         ENV["SCAN_INCLUDE_SIMULATOR_LOGS"] = true.to_s
       end
 
-      def self.setup_keychain
+      def self.setup_keychain(params)
         unless ENV["MATCH_KEYCHAIN_NAME"].nil?
           UI.message "Skipping Keychain setup as a keychain was already specified"
           return
         end
 
+        password = ""
+        if params[:random_password] # random password can be created optionally
+          password = SecureRandom.base64(12)
+        end
+
         keychain_name = "fastlane_tmp_keychain"
         ENV["MATCH_KEYCHAIN_NAME"] = keychain_name
-        ENV["MATCH_KEYCHAIN_PASSWORD"] = ""
+        ENV["MATCH_KEYCHAIN_PASSWORD"] = password
```

Comment:
The password is set here but not actually used later ([line 49](https://github.com/fastlane/fastlane/blob/2b2e008502deac13210f06b33afe111beb02eb04/fastlane/lib/fastlane/actions/setup_circle_ci.rb#L49))